overlay.go

```diff
@@ -490,6 +494,33 @@ func ReadAndMergeBundleData(sources ...BundleDataSource) (*BundleData, error) {
 	return base.Data, nil
 }
 
+// resolveOverlayPresenceFields exists because we expose an internal bundle
+// representation of a type out to the consumers of the library. This means it
+// becomes very difficult to know what was re-written during the normalisation
+// phase, without telling downstream consumers.
+//
+// The following attempts to guess when a normalisation has occurred, but the
+// presence field map is out of sync with the new changes.
+func resolveOverlayPresenceFields(base *BundleDataPart) {
+	applications := base.PresenceMap.forField("applications")
+	if len(applications) == 0 {
+		return
+	}
+	for name, app := range base.Data.Applications {
+		if !applications.fieldPresent(name) {
+			continue
+		}
+
+		presence := applications.forField(name)
+		// If the presence map contains scale, but doesn't containt num_units
+		// and if the app.Scale_ has be set to zero. We can then assume that a
+		// normalistion has occurred.
+		if presence.fieldPresent("scale") && !presence.fieldPresent("num_units") && app.Scale_ == 0 && app.NumUnits > 0 {
```

Comment:
I wonder whether there is a check somewhere `if scale > 0 && num_units >0 { // error }`. You shouldn't be specifying both at the same time right?

Reply:
Oh, just saw your test below. I think we should add the check as a drive-by. Otherwise, there is an ambiguity in the actual value for num units that must be resolved by the user.